packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx

```diff
@@ -0,0 +1,119 @@
+import {type ListenEvent, type ListenOptions} from '@sanity/client'
+import {useCallback, useEffect, useMemo, useState} from 'react'
+import {catchError, of} from 'rxjs'
+import {
+  type BundleDocument,
+  DEFAULT_STUDIO_CLIENT_OPTIONS,
+  getBundleSlug,
+  getPublishedId,
+  useBundles,
+  useClient,
+} from 'sanity'
+
+export interface DocumentPerspectiveProps {
+  documentId: string
+}
+
+export interface DocumentPerspectiveState {
+  data: BundleDocument[] | null
+  error?: Error
+}
+
+const LISTEN_OPTIONS: ListenOptions = {
+  events: ['welcome', 'mutation', 'reconnect'],
+  includeResult: true,
+  visibility: 'query',
+}
+
+/**
+ * Fetches the document versions for a given document
+ * @param props - document Id of the document (might include bundle slug)
+ * @returns - data: document versions, loading, errors
+ * @hidden
+ * @beta
+ */
+export function useDocumentVersions(props: DocumentPerspectiveProps): DocumentPerspectiveState {
```

Comment:
could this be replaced by the `useListener` hook? It seems to handle the same logic.

Reply:
Or for `listenQuery` ? And maybe we could update the projection to get only the id of the documents?